numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -3832,6 +3832,58 @@ test_interrupt(PyObject *NPY_UNUSED(self), PyObject *args)
     return PyInt_FromLong(a);
 }
 
+// Testing the utilites of the CPU dispatcher
+#ifndef NPY_DISABLE_OPTIMIZATION
+    #include "einsum.dispatch.h"
+#endif
+NPY_CPU_DISPATCH_DECLARE(extern const char *einsum_dispatch_var)
+NPY_CPU_DISPATCH_DECLARE(const char *einsum_dispatch_func, (void))
+NPY_CPU_DISPATCH_DECLARE(void einsum_dispatch_attach, (PyObject *list))
+
+static PyObject *
+test_einsum_dispatch(PyObject *NPY_UNUSED(dummy), PyObject *NPY_UNUSED(dummy2))
+{
+    const char *highest_func, *highest_var;
+    NPY_CPU_DISPATCH_CALL(highest_func = einsum_dispatch_func, ())
+    NPY_CPU_DISPATCH_CALL(highest_var  = einsum_dispatch_var)
+    const char *highest_func_xb = "nobase", *highest_var_xb = "nobase";
+    NPY_CPU_DISPATCH_CALL_XB(highest_func_xb = einsum_dispatch_func, ())
+    NPY_CPU_DISPATCH_CALL_XB(highest_var_xb  = einsum_dispatch_var)
+
+    PyObject *dict = PyDict_New(), *item;
+    if (dict == NULL) {
+        return NULL;
+    }
+    item = PyUnicode_FromString(highest_func);
+    if (item == NULL || PyDict_SetItemString(dict, "func", item) < 0) {
+        goto err;
+    }
+    item = PyUnicode_FromString(highest_var);
+    if (item == NULL || PyDict_SetItemString(dict, "var", item) < 0) {
+        goto err;
+    }
+    item = PyUnicode_FromString(highest_func_xb);
+    if (item == NULL || PyDict_SetItemString(dict, "func_xb", item) < 0) {
+        goto err;
+    }
+    item = PyUnicode_FromString(highest_var_xb);
+    if (item == NULL || PyDict_SetItemString(dict, "var_xb", item) < 0) {
+        goto err;
+    }
+    item = PyList_New(0);
+    if (item == NULL || PyDict_SetItemString(dict, "all", item) < 0) {
+        goto err;
+    }
+    NPY_CPU_DISPATCH_CALL_ALL(einsum_dispatch_attach, (item))
+    if (PyErr_Occurred()) {
+        goto err;
+    }
+    return dict;
+err:
+    Py_XDECREF(item);
+    Py_DECREF(dict);
+    return NULL;
+}
```

Comment:
this should go to one of the test extension modules, not to the main one. `_umath_tests` would be a good place.